File: src/test/java/io/sdata/core/route/RouterTest.java
package io.sdata.core.route;

import org.junit.Test;

import static org.junit.Assert.*;

public class RouterTest {

    @Test
    public void testAddRoute() throws Exception {
        Router<Object> router = new Router<>();
        router.addRoute("http://www.amazon.com/s/:*", "1");
        router.addRoute("http://www.amazon.com/sp/:*", "2");
        assertEquals(2, router.routes().size());
    }

    @Test
    public void testRemovePath() throws Exception {
        Router<Object> router = new Router<>();
        router.addRoute("http://www.amazon.com/s/:*", "1");
        router.addRoute("http://www.amazon.com/sp/:*", "2");
        router.removePath("http://www.amazon.com/sp/:*");
        router.removePath("http://www.amazon.com/sp/:*");
        assertEquals(1, router.routes().size());
    }

    @Test
    public void testRoute() throws Exception {
        Router<Object> router = new Router<>();
        router.addRoute("http://www.amazon.com/s/:*", "1");
        router.addRoute("http://www.amazon.com/sp/:*", "2");
        RouteResult<Object> result = router.route("http://www.amazon.com/s/ref=lp_7147441011_ex_n_1?rh=n%3A7141123011%2Cn%3A10445813011");
        assertTrue("1".equals(result.target()));
        result = router.route("http://www.amazon.com/sp/ref=lp_7147441011_ex_n_1?rh=n%3A7141123011%2Cn%3A10445813011");
        assertTrue("2".equals(result.target()));
    }
}

File: src/test/scala/com/sfetcher/core/route/RouterTest.scala
package com.sfetcher.core.route

import org.scalatest.FreeSpec

/**
  * Created by dejun on 26/2/16.
  */
class RouterTest extends FreeSpec {

  "RouterTest" - {
    "removePath" in {
      val router: Router[AnyRef] = new Router[AnyRef]
      router.addRoute("http://www.amazon.com/s/*", "1")
      router.addRoute("http://www.amazon.com/sp/*", "2")
      router.removePath("http://www.amazon.com/sp/*")
      router.removePath("http://www.amazon.com/sp/*")
      assert(router.routes.size === 1)
    }

    "routes" in {
      val router: Router[AnyRef] = new Router[AnyRef]
      router.addRoute("http://www.amazon.com/s/*", "1")
      router.addRoute("http://www.amazon.com/sp/*", "2")
      var result: RouteResult[AnyRef] = router.route("http://www.amazon.com/s/ref=lp_7147441011_ex_n_1?rh=n%3A7141123011%2Cn%3A10445813011")
      assert(result.target === "1")
      result = router.route("http://www.amazon.com/sp/ref=lp_7147441011_ex_n_1?rh=n%3A7141123011%2Cn%3A10445813011")
      assert(result.target === "2")
    }

    "addRoute" in {
      val router: Router[AnyRef] = new Router[AnyRef]
      router.addRoute("http://www.amazon.com/s/*", "1")
      router.addRoute("http://www.amazon.com/sp/*", "2")
      assert(router.routes.size == 2)
    }

  }
}
